docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,108 @@
+Cucumber Core is the [inner hexagon](https://en.wikipedia.org/wiki/Hexagonal_architecture_(software)) for the [Ruby flavour of Cucumber](https://github.com/cucumber/cucumber-ruby). It contains the core domain logic to execute Cucumber features. It has no user interface, just a Ruby API.
+
+## Overview
+
+The entry-point is a single method on the module `Cucumber::Core` called
+`#execute`. Here's what it does:
+
+1. Parses the plain-text Gherkin documents into an **AST**
+2. Compiles the AST down to **test cases**
+3. Passes the test cases through any **filters**
+4. Executes the test cases, emitting **events** as it goes
+
+### The AST
+
+The Abstract Syntax Tree or AST is an object graph that represents the Gherkin documents you've passed into the core. Things like `Feature`, `Scenario` and `ExamplesTable`.
+
+These are immutable value objects.
+
+### Test cases
+
+Your gherkin might contain scenarios, as well as examples from tables beneath a scenario outline.
```

Comment:
This could be capitalized, I guess, like on the line before.

Reply:
You mean "Gherkin"? Yes, I'll fix this :)